numcodecs/tests/common.py

```diff
@@ -215,17 +215,15 @@ def check_backwards_compatibility(codec_id, arrays, codecs, precision=None, pref
             # one time encode and save array
             if not os.path.exists(enc_fn):  # pragma: no cover
                 enc = codec.encode(arr)
+                enc = ensure_bytes(enc)
                 with open(enc_fn, mode='wb') as ef:
                     ef.write(enc)
 
             # load and decode data
             with open(enc_fn, mode='rb') as ef:
                 enc = ef.read()
                 dec = codec.decode(enc)
-                if isinstance(dec, np.ndarray):
-                    dec_arr = dec.reshape(-1, order='A')
-                else:
-                    dec_arr = ensure_ndarray(dec)
+                dec_arr = ensure_ndarray(dec).reshape(-1, order='A')
```

Comment:
For the record, this is just a simplification, not needed to address the issue at hand.